src/sentry/tasks/on_demand_metrics.py

```diff
@@ -49,6 +49,10 @@ def _get_widget_processing_batch_key() -> str:
     return "on-demand-metrics:widgets:currently-processing-batch"
 
 
+def _get_project_for_query_cache_key(organization: Organization) -> str:
+    return f"project_for_query:{organization.id}"
```

Comment:
Could we give this a more descriptive cachekey? maybe something like
```suggestion
    return f"on_demand_project_spec:{organization.id}"
```